### ISSUE TITLE
TypeScript nested locals are reported as exported module symbols

### ISSUE THREAD

**Issue by styler-ai** [styler-ai/ProjectAtlas#143]
## Summary

ProjectAtlas marked TypeScript symbols nested inside an exported function as `exported: true` and included those private locals in the file-level exports list.

This was reproduced against a representative large application corpus, not the ProjectAtlas source tree.

## Reproduction Shape

Index a TypeScript file containing an exported function with nested local values/functions/types:

```ts
export function useProductFixtureState(): CatalogControllerState {
  type LocalMode = 'all' | 'filtered';
  const localCache = computed(...);
  const itemMap = computed(...);
  const passesFilters = (item: CatalogFixtureItem): boolean => { ... };
}
```

Then inspect the file summary or symbols:

```text
projectatlas summary fixtures/catalog/useProductFixtureState.ts --limit 25
projectatlas symbols list --file fixtures/catalog/freshness.ts --limit 40
```

## Observed

Nested private locals under an exported function were marked as exported and appeared in the file-level `exports[...]` list.

## Expected

`exported` should reflect actual TypeScript module export status, not visibility inherited from an exported parent function.

Nested symbols should keep their `parent` containment, but private locals such as `localCache`, `itemMap`, `passesFilters`, and local type aliases should be `exported: false` and absent from the module-level exports list.

## Impact

Agents using ProjectAtlas summaries for public API maps get an inflated module surface. Private implementation details look safe to import or treat as stable API.
